### PR TITLE
Commonjs bundle - take 2

### PR DESCRIPTION
**Pull request by Jonic**
* Includes package-lock.json - "This file is intended to be committed into source repositories" (https://docs.npmjs.com/files/package-lock.json)
* Add automatic script compilation when installed via `npm` - automatically generates `scratchblocks.min.js`
* Adds a `main` config definition to `package.json` so that it can be imported by Node
* Auto lints JS and `package.json` on commit
* `package.json` and `.gitignore` have been tidied a bit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,42 +1,59 @@
...
+  "directories": {
+    "test": "tests"
+  },
+  "scripts": {
+    "build": "browserify browser.js -o build/scratchblocks.min.js -t [ babelify --presets [ es2015 babili ] ]",
+    "fmt": "prettier --write *.js syntax/*.js scratch2/*.js",
+    "postinstall": "npm run build -- -s Scratchblocks",
+    "lint:staged": "lint-staged",
+    "start": "beefy browser.js:bundle.js --live --port 8000",
+    "test": "jest"
```
Comment (tjvr) [inline package.json]: How does this work, since `build/` isn't committed? :)

Comment (Jonic) [reply package.json]: The `build` here refers to the npm script, not the directory. So it's just like when you run `npm run build` in Terminal. I've actually got another branch in the works that updates the config for building stuff a little, and that DOES include the build directory in the tree, so we can chat about that then.

For now if you want to see this in action, run `npm i` in the root of the repo and you'll see the postinstall command run, and the `scratchblocks.min.js` file created in the `build` directory :)

Comment (Jonic) [reply package.json]: I've been testing this on my local copy of our projects site, where `import scratchblocks from 'scratchblocks'` works fine, but I'll confirm that `require('scratchblocks')` also still works. Can't see why it wouldn't, but will make absolutely sure anyway :)

Comment (tjvr) [reply package.json]: Ah, I missed the `postinstall` command! Gotcha. :)

**Pull request merged**